app/src/main/java/com/omkarmoghe/pokemap/views/LoginActivity.java

```diff
@@ -48,13 +50,22 @@
     private GoogleManager.LoginListener mGoogleLoginListener;
 
     private String mDeviceCode;
+    private PokemapAppPreferences mPref;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_login);
+
         mNianticManager = NianticManager.getInstance();
         mGoogleManager = GoogleManager.getInstance();
+        mPref = new PokemapSharedPreferences(this);
+        
+        if (mPref.isUsernameSet() || mPref.isPasswordSet()) {
```

Comment:
Good point. Made the change. Some variation of this logic will be final build worthy IMO, but my main comment about login not being ready is based on not using tokens and always logging in each time the app is started.